benches/basic_incr.rs

```diff
@@ -0,0 +1,37 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use prima_datadog::{
+    configuration::{Country, PrimaConfiguration},
+    Datadog,
+};
+
+fn incr_benchmark(c: &mut Criterion) {
+    let configuration = PrimaConfiguration::new(
+        "0.0.0.0:1234",
+        "0.0.0.0:0",
+        "prima_datadog_benchmarks",
+        "dev".parse().unwrap(),
+    )
+    .with_country(Country::It)
+    .with_tag_warn_threshold(21);
+    // 20 test tags to simulate a normal to heavily tagged metric
+    let tags = (0..20).map(|i| format!("tag_{}", i)).collect::<Vec<_>>();
+    Datadog::init(configuration).unwrap();
+    c.bench_function("incr_benchmark", |b| {
+        b.iter(|| {
+            // Note: clone here is ok since we only care about /relative/ perf here
+            Datadog::incr("test", tags.clone());
+        });
+    });
+}
+
+fn incr_with_too_many_tags(c: &mut Criterion) {
+    let tags = (0..22).map(|i| format!("tag_{}", i)).collect::<Vec<_>>();
+    c.bench_function("incr_with_too_many_tags", |b| {
+        b.iter(|| {
+            Datadog::incr("test", tags.clone());
+        });
+    });
+}
+
+criterion_group!(benches, incr_benchmark, incr_with_too_many_tags);
+criterion_main!(benches);
```

Comment:
1) On line 15 I set the warning threshold to 21, so the difference between 20 and 22 is that one is under that threshold and the other isn't (basically, the second benchmark hits the atomic compare exchange, but both pass 20 tags with each metric in order to have fairly similar workload with respect to hashing the tags etc)

2) Because of how criterion works, any other function would look like another benchmark. The order of benching is set by the macros and stable - I could break it out but it doesn't gain us anything, and unfortunately the benchmarks are interdependent here because of the OnceCell the whole library uses (and doesn't expose a way to reset) - I would like each benchmark to do it's own setup but because criterion doesn't have access to library methods I would have to expose a way to reset the OnceCell to all users, which doesn't seem like a good idea.

Reply:
I completely missed line 15! Now I understand, but I still feel having Datadog initialized in a function and then implicitly used in another is quite confusing, if there's a way to change this to make it more explicit/clean I would do it!

Also, what are we trying to benchmark?

Reply:
I can break it out into a separate "benchmark" called something like "setup", is that ok?

Reply:
A setup function would make sense I think!